src/segment_reshape/map_tool/segment_reshape_tool.py

```diff
@@ -158,6 +166,21 @@ def _handle_mouse_click_event(
                 success=True,
             )
 
+    def _handle_key_event(self, key: Qt.Key) -> None:
+        if self.reshape_mode is True:
+            if key == Qt.Key_Escape or len(self.added_points) == 0:
```

Comment:
would this actually match any key press? for example both if esc is pressed, or any-key with empty points list would be true here.

Reply:
Ash, I guess you are right. I was trying to catch the situation where the user presses backspace in the situation where he has selected the segment to reshape but hasn't yet clicked any vertices. I'll change the second condition to only concern situations when backspace has been clicked in addition to the empty list of selected vertices.